docs/design/kademlia_audit_gating.md

```diff
@@ -0,0 +1,94 @@
+
+4. Modify FindNear to return n XOR-closest nodes from the antechamber in addition to those already returned from the routing table.
+
+5. On deployment, avoid complete erasure of the network's routing tables.
+
+
+## Terminology
+**Routing Table Antechamber** - *XOR-ordered temporary holding place for unverified storagenodes*
+
+**Node Neighborhood** - *The k-closest nodes to self where distance is measured by XOR. A node is within the node neighborhood if it is closer than the furthest node in the neighborhood. The vetted node neighborhood is the k-closest nodes that are currently in the Routing Table.*
+
+[TODO Node Neighborhood Diagram]
+
+## Design
+
+1. Satellite Signatures for Node Verification
+    - Identities can sign messages already
+    - Create a Message (protobuf) to sign that Satellite C says Node B has been vetted
+    - Get info about the Node's difficulty and disk space to add to the message
+
+2. Trusted Satellites List
+    - Create Whitelist/blacklist with an abstraction layer for trusted/untrusted Satellites
+    - These lists will live on each Node
+
+3. Routing Table Antechamber
+    - XOR ordered data structure (perhaps an ordered slice)
+    - A node can be added if it would be within the vetted node neighborhood
+    - Once a node has been verified, it broadcasts its new status to try to join routing tables
+    - A Node may enter a Routing Table directly if at first contact it is already verified by a trusted Satellite.  
+    - A Node may be removed from a Routing Table if on bucket refresh it no longer meets qualifications
+    - Since the antechamber may only contain nodes that would be part of the node neighborhood, if the network grows, the space in the neighborhood will shrinks so we must remove antechamber nodes that no longer fit in the neighborhood.
+  
+4. FindNear
+    - should return n XOR-closest nodes from the antechamber in addition to its current behavior
+    - During processes like Bootrapping or Kademlia Lookups, we should only call FindNear on verified nodes, not on those that are in the antechamber.
+
+5. Progressively migrate nodes from the current Routing Table into antechamber until verified. 
+    - Deploy vetting and signing first
+    - Deploy antechamber and nodes will naturally shift
+
+
+## Rationale
+
+To help prevent Sybill Attacks where bad Nodes fill the routing tables and push out good Nodes.
+
+## Implementation
+
+1. Implement [Satellite Signatures for Node Verification](https://storjlabs.atlassian.net/browse/V3-1726)
+
+2. Deploy step 1 to production
+
+3. Make [Trusted Satellites List](https://storjlabs.atlassian.net/browse/V3-1727)
+
+4. Create [Routing Table Antechamber](https://storjlabs.atlassian.net/browse/V3-1728)
+
+5. [Update FindNear](https://storjlabs.atlassian.net/browse/V3-1729)
+
+## Open issues 
+
+Q: Should closer buckets to self get refreshed more frequently?
+
+A: ?
+
+Q: If a node is removed from the Routing Table, should it be added back to the antechamber?
```

Comment:
maybe it depends on if the node got removed because of not having enough disk space or high enough ID difficulty vs. if it got removed because it was too far away? not sure, it seems like all of those factors could change, so maybe it's okay to let it back in? maybe if a node was disqualified for audit failures then it shouldn't be let back in.